### PR TITLE
PR test&issue

### PR DESCRIPTION
**Pull request by 0xenj**
- Setup test file, error file
- #42 
- #44 
- #45

### CODE REVIEW

File: contracts/SimpleNftMarketplace.sol [hunk 1]
```diff
@@ -60,25 +60,30 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
 
   // Admin
   function changeSupportedContract(address contractAddress, bool isSupported) external onlyAdmin returns (bool success) {
-    return false;
+    _changeSupportedContract(contractAddress);
+    return true;
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:64]: _changeSupportedContract(contractAddress) already return bool
so change for
```solidity
return _changeSupportedContract(contractAddress);
```

File: contracts/SimpleNftMarketplace.sol [hunk 2]
```diff
@@ -60,25 +60,30 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
   }
 
-  function changeTransactionFee(uint32 transactionFee) external onlyAdmin returns (bool success) {
-    return false;
+  function changeTransactionFee(uint32 newTransactionFee) external onlyAdmin returns (bool success) {
+    _changeTransactionFee(newTransactionFee);
+    return true;
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:69]: _changeTransactionFee(newTransactionFee) already return bool
so change for
```solidity
return _changeTransactionFee(newTransactionFee);
```

File: contracts/SimpleNftMarketplace.sol [hunk 3]
```diff
@@ -60,25 +60,30 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
   }
 
   // Treasury
   function withdrawTransactionFee() external onlyTreasury returns (bool success) {
-    return false;
+    _withdrawTransactionFee();
+    return true;
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:75]: _withdrawTransactionFee() already return bool
so change for
```solidity
return _withdrawTransactionFee();
```

File: contracts/SimpleNftMarketplace.sol [hunk 4]
```diff
@@ -60,25 +60,30 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
   }
 
   // Moderator
-  function blacklistToken(address tokenContract, uint256 tokenId) external onlyModerator returns (bool success) {
-    return false;
+  function blacklistToken(address tokenContract, uint256 tokenId, bool isBlackListed) external onlyModerator returns (bool success) {
+    _blacklistToken(tokenContract, tokenId, isBlackListed);
+    return true;
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:81]: _blacklistToken(tokenContract, tokenId, isBlackListed) already return bool
so change for
```solidity
return _blacklistToken(tokenContract, tokenId, isBlackListed);
```

File: contracts/SimpleNftMarketplace.sol [hunk 5]
```diff
@@ -60,25 +60,30 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
   }
 
-  function blacklistUser(address userAddress) external onlyModerator returns (bool success) {
-    return false;
+  function blacklistUser(address userAddress, bool set) external onlyModerator returns (bool success) {
+    _blacklistUser(userAddress, set);
+    return true;
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:86]: _blacklistUser(userAddress, set) already return bool
so change for
```solidity
return _blacklistUser(userAddress, set);
```

File: contracts/SimpleNftMarketplace.sol [hunk 6]
```diff
@@ -92,11 +97,23 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
     return _listings[listingId].buyer == address(0) && _listings[listingId].seller != address(0);
   }
 
-  function isBlacklistedUser(address userAddress) external view returns (bool isBlacklisted) {}
+  function isBlacklistedUser(address userAddress) external view returns (bool isBlacklisted) {
+    if (_isBlacklistedUser(userAddress) == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:103]: _isBlacklistedUser(userAddress)  already return bool
so remove the if and just add
```solidity
return _isBlacklistedUser(userAddress);
```

File: contracts/SimpleNftMarketplace.sol [hunk 7]
```diff
@@ -92,11 +97,23 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
+  }
 
-  function isBlacklistedToken(address tokenContract, uint256 tokenId) external view returns (bool isBlacklisted) {}
+  function isBlacklistedToken(address tokenContract, uint256 tokenId) external view returns (bool isBlacklisted) {
+    if (_isBlacklistedToken(tokenContract, tokenId) == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:109]: _isBlacklistedToken(tokenContract, tokenId)  already return bool
so remove the if and just add
```solidity
return _isBlacklistedToken(tokenContract, tokenId);
```

File: contracts/SimpleNftMarketplace.sol [hunk 8]
```diff
@@ -92,11 +97,23 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
+  }
 
-  function isSupportedContract(address tokenContract) external view returns (bool isSupported) {}
+  function isSupportedContract(address tokenContract) external view returns (bool isSupported) {
+    if (_isSupportedContract(tokenContract) == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:115]: _isSupportedContract(tokenContract) already return bool
so remove the if and just add
```solidity
return _isSupportedContract(tokenContract);
```

File: contracts/abstracts/Controlable.sol [hunk 9]
```diff
@@ -80,6 +80,24 @@ abstract contract Controlable is AccessControlUpgradeable {
     }
   }
 
+  function _isBlacklistedUser(address userAddress) internal view returns (bool isBlacklisted) {
+    if (blacklistUser[userAddress] == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/abstracts/Controlable.sol:86]: blacklistUser[userAddress] already equal bool
so remove the if and just add
```solidity
return blacklistUser[userAddress];
```

File: contracts/abstracts/Controlable.sol [hunk 10]
```diff
@@ -80,6 +80,24 @@ abstract contract Controlable is AccessControlUpgradeable {
...
+  }
+
+  function _isBlacklistedToken(address tokenContract, uint256 tokenId) internal view returns (bool isBlacklisted) {
+    if (blacklistToken[tokenContract][tokenId] == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/abstracts/Controlable.sol:92]: blacklistToken[tokenContract][tokenId] already equal bool
so remove the if and just add
```solidity
return blacklistToken[tokenContract][tokenId];
```

File: contracts/abstracts/Controlable.sol [hunk 11]
```diff
@@ -80,6 +80,24 @@ abstract contract Controlable is AccessControlUpgradeable {
...
+  }
+
+  function _isSupportedContract(address tokenContract) internal view returns (bool isSupported) {
+    if (supportedContracts[tokenContract] == true) {
+      return true;
+    }
```
Comment (marc-aurele-besner) [inline contracts/abstracts/Controlable.sol:98]: supportedContracts[tokenContract] already equal bool
so remove the if and just add
```solidity
return supportedContracts[tokenContract];
```

**Review by marc-aurele-besner — CHANGES_REQUESTED** [review]

File: contracts/SimpleNftMarketplace.sol [hunk 12]
```diff
@@ -92,11 +92,17 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
     return _listings[listingId].buyer == address(0) && _listings[listingId].seller != address(0);
   }
 
-  function isBlacklistedUser(address userAddress) external view returns (bool isBlacklisted) {}
+  function isBlacklistedUser(address userAddress) external view returns (bool isBlacklisted) {
+    return (_isBlacklistedUser(userAddress) == true);
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:96]: _isBlacklistedUser(userAddress)  already return true or false, no need to add == true

File: contracts/SimpleNftMarketplace.sol [hunk 13]
```diff
@@ -92,11 +92,17 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
+  }
 
-  function isBlacklistedToken(address tokenContract, uint256 tokenId) external view returns (bool isBlacklisted) {}
+  function isBlacklistedToken(address tokenContract, uint256 tokenId) external view returns (bool isBlacklisted) {
+    return (_isBlacklistedToken(tokenContract, tokenId) == true);
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:100]: same as above

File: contracts/SimpleNftMarketplace.sol [hunk 14]
```diff
@@ -92,11 +92,17 @@ contract SimpleNftMarketplace is ListingManager, ValidateSignature {
...
+  }
 
-  function isSupportedContract(address tokenContract) external view returns (bool isSupported) {}
+  function isSupportedContract(address tokenContract) external view returns (bool isSupported) {
+    return (_isSupportedContract(tokenContract) == true);
```
Comment (marc-aurele-besner) [inline contracts/SimpleNftMarketplace.sol:104]: same as above

**Review by marc-aurele-besner — CHANGES_REQUESTED** [review]

**Comment by 0xenj** [comment]
yep, my bad for this

**Review by marc-aurele-besner — APPROVED** [review]

File: contracts/abstracts/ListingManager.sol [hunk 15]
```diff
@@ -31,6 +31,9 @@ abstract contract ListingManager is Controlable {
   }
 
   function _createListing(address tokenContract, uint256 tokenId, uint256 salePrice, address seller) internal returns (uint256 listingId) {
+    require(_isBlacklistedUser(seller) == false, 'User is blacklisted');
```
Comment (marc-aurele-besner) [inline contracts/abstracts/ListingManager.sol:34]: You can simplify this line (and the 2 others by replacing it by
```solidity
require(!_isBlacklistedUser(seller), 'User is blacklisted');
```
Also please add the name of the contract in the revert message, so we can more easily trace back where the error originate, like
```txt
ListingManager: User is blacklisted
```

**Review by marc-aurele-besner — CHANGES_REQUESTED** [review]

**Review by marc-aurele-besner — APPROVED** [review]

**Pull request merged**